jdisc_http_service/src/main/java/com/yahoo/jdisc/http/server/jetty/HealthCheckProxyHandler.java

```diff
@@ -124,21 +115,86 @@ protected void doStop() throws Exception {
         super.doStop();
     }
 
+    private static class ProxyRequestTask implements Runnable {
+
+        final AsyncContext asyncContext;
+        final ProxyTarget target;
+        final HttpServletResponse servletResponse;
+        final ServletOutputStream output;
+
+        ProxyRequestTask(AsyncContext asyncContext, ProxyTarget target, HttpServletResponse servletResponse, ServletOutputStream output) {
+            this.asyncContext = asyncContext;
+            this.target = target;
+            this.servletResponse = servletResponse;
+            this.output = output;
+        }
+
+        @Override
+        public void run() {
+            StatusResponse statusResponse = target.requestStatusHtml();
+            servletResponse.setStatus(statusResponse.statusCode);
+            if (statusResponse.contentType != null) {
+                servletResponse.setHeader("Content-Type", statusResponse.contentType);
+            }
+            output.setWriteListener(new WriteListener() {
+                @Override
+                public void onWritePossible() throws IOException {
+                    if (output.isReady()) {
+                        if (statusResponse.content != null) {
+                            output.write(statusResponse.content);
+                        }
+                        asyncContext.complete();
+                    }
+                }
+
+                @Override
+                public void onError(Throwable t) {
+                    log.log(Level.FINE, t, () -> "Failed to write status response: " + t.getMessage());
+                    asyncContext.complete();
+                }
+            });
+        }
+    }
+
     private static class ProxyTarget implements AutoCloseable {
         final int port;
         final Duration timeout;
         final SslContextFactory.Server sslContextFactory;
         volatile CloseableHttpClient client;
+        volatile StatusResponse lastResponse;
 
         ProxyTarget(int port, Duration timeout, SslContextFactory.Server sslContextFactory) {
             this.port = port;
             this.timeout = timeout;
             this.sslContextFactory = sslContextFactory;
         }
 
-        CloseableHttpResponse requestStatusHtml() throws IOException {
-            return client()
-                    .execute(new HttpGet("https://localhost:" + port + HEALTH_CHECK_PATH));
+        StatusResponse requestStatusHtml() {
+            StatusResponse response = lastResponse;
+            if (response != null && !response.isExpired()) {
+                return response;
+            }
+            StatusResponse statusResponse = getStatusResponse();
+            lastResponse = statusResponse;
+            return statusResponse;
```

Comment:
```suggestion
            return lastResponse = getStatusResponse();
```
I find this easier to read, but personal preference I guess :)

Reply:
Done :)